### PR TITLE
Update: Warn on directives in line comments (refs eslint/rfcs#34)

### PR DESCRIPTION
**Pull request by captbaritone**
**Note:** This RFC is still in Final Commenting phase, but I had this PR ready so I figured I'd open it.

**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[X] Add something to the core
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

RFC #34 [Allow All Directives in Line Comments](https://github.com/eslint/rfcs/pull/34) aims to allow all directives in line comments (`// Line comment`). This will need to wait for a major version bump since code bases may include line comments which will accidentally parse as directives.

To mitigate the impact of these accidental directives we plan to ship a minor version which warns when line comments contain ESLint directives which are only allowed in block comments. This PR add that behavior.

This will help in two ways:

1. Users who accidentally try to use a block level only directive in a line comment will get a helpful error message rather than silently ignoring their directive.
2. Any users who upgrade to this minor version before the major version which begins respecting all directives in both types of comments will avoid the possibility of accidentally interpreting comments which were not intended to be directives as directives.

**Is there anything you'd like reviewers to focus on?**

Should we provide a link in the warning which gives more information as to why this these comments are now causing errors? Is there any convention of including links in lint errors?

### CODE REVIEW

File: tests/lib/rules/line-comment-position.js [hunk 1]
```diff
@@ -28,9 +28,6 @@ ruleTester.run("line-comment-position", rule, {
         "1 + 1; /* eslint-enable */",
         "1 + 1; // eslint-disable-line",
         "// eslint-disable-next-line\n1 + 1;",
-        "1 + 1; // global MY_GLOBAL, ANOTHER",
-        "1 + 1; // globals MY_GLOBAL: true",
-        "1 + 1; // exported MY_GLOBAL, ANOTHER",
```
Comment (captbaritone) [inline tests/lib/rules/line-comment-position.js]: These lines should actually still be valid from the perspective of the `line-comment-position` rule but we now create a warning when parsing them. Is there a sensible way to keep these assertions? Maybe I should add them to the `invalid` section but somehow specify that we only expect them to error with the new error and not this rule's error?

**Comment by captbaritone** [comment]
Commit message check still seems unhappy. Should I file an issue on eslint/eslint for this first phase of the RFC?

**Comment by platinumazure** [comment]
@captbaritone Try editing the PR title to something like this:

```
Update: Warn on line comments with directives that are only allowed in block comments (fixes eslint/rfcs#34)
```

Note that this might still be too long overall.

**Comment by mysticatea** [comment]
I'm wondering if that's a breaking change because it increases warnings for a new feature.

**Comment by captbaritone** [comment]
@mysticatea would you mind raising that concern on the RFC issue? Seems like that would be the best place to make that judgement call.

**Comment by nzakas** [comment]
**TSC Summary**: This PR implements the warning originally described in eslint/rfcs#34. This functionality was removed prior to finalizing the RFC. Initially, this was a first step to warn people of directive-like comments that were not currently supported before ultimately supporting them.

**TSC Question**:  Do we want to merge this PR or just go straight to implementing eslint/rfcs#34?

**Comment by btmills** [comment]
We're sorry for losing track of this for so long. RFC34 ended up diverging from this PR, so rather than merging this PR as an intermediate step, we'd rather have a single PR to do all of RFC34. If you're still willing to submit a PR to do that, we'll be sure to respond to that one sooner than 18 months :grimacing:

**Pull request closed without merging**